src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java

```diff
@@ -450,7 +457,16 @@ public static boolean endsInPeriod(List<CobolLine> lines) {
      * This will make sure to add or remove any referenced field within the structure, based on
      * the level of said field within the structure.
      */
-    public static TreeMap<Integer,String> updateCurrentDataStructure(CobolLine line, TreeMap<Integer, String> currentHierarchy) {
+    public static TreeMap<Integer,String> updateCurrentDataStructure(List<CobolLine> currentStatement, TreeMap<Integer, String> currentHierarchy) {
```

Comment:
I think this method is starting to be a bit long. Could some parts be refactored into other methods?